src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -201,6 +201,13 @@ public void RegisterForPostInitialization(Action<GeneratorPostInitializationCont
                 throw new InvalidOperationException(string.Format(CodeAnalysisResources.Single_type_per_generator_0, typeName ?? typeof(T).Name));
             }
         }
+
+        internal class CallbackHolder
```

Comment:
>CallbackHolder [](http://example.com/codeflow?start=23&length=14)

Why introduce the level of indirection rather than defining these fields on `GeneratorInitializationContext` directly? Are we passing around a `CallbackHolder` instance anywhere?

Reply:
The struct get passed to `ISourceGenerator.Initialize` where the generator then calls the set methods. Previously we had a builder type that held onto them, but given that that's no longer used I just moved them here. Its ugly but its just for back compat.